### ISSUE TITLE
doc/go1.16: document go command changes for Go 1.16

### ISSUE THREAD

**Issue by dmitshur** [golang/go#42893]
As of filing this issue, the [Go 1.16 draft release notes](https://tip.golang.org/doc/go1.16) contained the following HTML:

```HTML
<h3 id="go-command">Go command</h3>

<p>
  TODO

  <!-- CL 237697: https://golang.org/cl/237697: cmd/go: error when -c or -i are used with unknown flags -->
  <!-- CL 255052: https://golang.org/cl/255052: cmd/go: default to GO111MODULE=on -->
  <!-- CL 266420: https://golang.org/cl/266420: yes (mention go help vcs): cmd/go: add GOVCS setting to control version control usage -->
  <!-- CL 244773: https://golang.org/cl/244773: cmd/go/internal/modload: drop requirements on excluded versions -->
</p>
```

And these TODOs:

```
TODO: write and link to section in golang.org/ref/mod
TODO: write and link to blog post
[...]
TODO: write and link to section in golang.org/ref/mod
TODO: write and link to tutorial or blog post
```

Per [golang.org/s/release](https://golang.org/s/release), it's a release blocker for Go 1.16 Beta 1 to have a complete draft of the eventual release notes, and so the TODO needs to be resolved.

A sequence of steps to resolve this issue may be:

1. Come up with a draft release note entry in this issue, then move the issue to NeedsFix state.
2. Send a CL with "doc/go1.16:" [prefix](https://go-review.googlesource.com/q/project:go+doc/go1.16) and include "For #40700. Fixes #{this issue}." in the body.

**Comment by bcmills** [comment]
@jayconrod, is this resolved by https://golang.org/cl/274438?

**Comment by jayconrod** [comment]
Yes, https://golang.org/cl/274438 removed all the TODOs listed here. I saw Alex's email to gophers, but i didn't see this issue.

@matloob I'll leave this issue open for now. Please close it after adding a note for `-overlay`.

**Comment by dmitshur** [comment]
CL 274438 has resolved all user-facing TODOs in the draft 1.16 release notes, so I think the rest of the work on this issue doesn't need to block beta 1.

**Comment by dmitshur** [comment]
@jayconrod Has the note for `-overlay` been added too (as you mentioned in https://github.com/golang/go/issues/42893#issuecomment-736784739), or is it okay not to have an open issue for that?

**Comment by jayconrod** [comment]
Oops, I'm sorry, https://go-review.googlesource.com/c/go/+/274714 is the CL adding the `-overlay` note, and it has not been merged yet, so reopening.

**Comment by dolmen** [comment]
Is there some reference documentation about the `-overlay` flag? The not-yet-published 1.16 release notes which mention `-overlay` doesn't link to documentation about the JSON file structure with example.

**Comment by dmitshur** [comment]
@dolmen The `-overlay` flag is documented in the [not-yet-published 1.16 cmd/go documentation](https://tip.golang.org/cmd/go/#hdr-Compile_packages_and_dependencies), search for "-overlay" on that page.
